websocket/src/websocket.cpp

```diff
@@ -799,20 +824,10 @@ static dmExtension::Result OnUpdate(dmExtension::Params* params)
             emscripten_websocket_set_onerror_callback(ws, conn, Emscripten_WebSocketOnError);
             emscripten_websocket_set_onclose_callback(ws, conn, Emscripten_WebSocketOnClose);
             emscripten_websocket_set_onmessage_callback(ws, conn, Emscripten_WebSocketOnMessage);
-            SetState(conn, STATE_CONNECTING);
 #else
-            dmSocket::Result sr;
-            int timeout = g_Websocket.m_Timeout;
-            dmConnectionPool::Result pool_result = dmConnectionPool::Dial(g_Websocket.m_Pool, conn->m_Url.m_Hostname, conn->m_Url.m_Port, g_Websocket.m_Channel, conn->m_SSL, timeout, &conn->m_Connection, &sr);
-            if (dmConnectionPool::RESULT_OK != pool_result)
-            {
-                CLOSE_CONN("Failed to open connection: %s", dmSocket::ResultToString(sr));
-                continue;
-            }
-            conn->m_Socket = dmConnectionPool::GetSocket(g_Websocket.m_Pool, conn->m_Connection);
-            conn->m_SSLSocket = dmConnectionPool::GetSSLSocket(g_Websocket.m_Pool, conn->m_Connection);
-            SetState(conn, STATE_HANDSHAKE_WRITE);
+            conn->m_ConnectionThread = dmThread::New((dmThread::ThreadStart)ConnectionWorker, 0x80000, conn, "WebSocketConnectionThread");
 #endif
+            SetState(conn, STATE_CONNECTING);
         }
         else if (STATE_CONNECTING == conn->m_State)
```

Comment:
Do we need to deal with our own timeout?
What happens if we call CLOSE_CONN, and the connection+thread is still in use?

Reply:
Good point. Skipping duplicate timeout check on all platforms but emscripten